projects/plugins/jetpack/_inc/client/security/waf.jsx

```diff
@@ -0,0 +1,299 @@
+							<Textarea
+								disabled={
+									! isWafActive ||
+									unavailableInOfflineMode ||
+									this.props.isSavingAnyOption( [
+										'waf',
+										'jetpack_waf_ip_list',
+										'jetpack_waf_ip_allow_list',
+									] )
+								}
+								name="jetpack_waf_ip_allow_list"
+								placeholder={ __( 'Example:', 'jetpack' ) + '\n12.12.12.1\n12.12.12.2' }
+								value={ this.state.jetpack_waf_ip_allow_list }
+								onChange={ this.handleIpListChange }
+							/>
+						</div>
+					</>
+				) }
+			</div>
+		);
+
+		const shareDataSettings = (
+			<div className="waf__settings__toggle-setting">
+				<CompactFormToggle
+					checked={ this.state.jetpack_waf_share_data }
+					disabled={
+						! isWafActive ||
+						unavailableInOfflineMode ||
+						this.props.isSavingAnyOption( [ 'waf', 'jetpack_waf_share_data' ] )
+					}
+					onChange={ this.handleShareDataToggleChange }
+				>
+					<span className="jp-form-toggle-explanation">
+						{ __( 'Share data with Jetpack', 'jetpack' ) }
+					</span>
+				</CompactFormToggle>
+				<InfoPopover
+					position="right"
+					screenReaderText={ __( 'Learn more', 'jetpack' ) }
+					className="waf__settings__share-data-popover"
+				>
+					{ createInterpolateElement(
+						__(
+							'Allow Jetpack to collect data to improve Firewall protection and rules. <a>Learn more</a>',
+							'jetpack'
+						),
+						{
+							a: (
+								<a
+									href="https://jetpack.com/support/privacy/"
+									target="_blank"
+									rel="noopener noreferrer"
+								/>
+							),
+						}
+					) }
+					<hr />
+					<a href="https://jetpack.com/support/privacy/" target="_blank" rel="noopener noreferrer">
+						{ __( 'Privacy information', 'jetpack' ) }
+					</a>
+				</InfoPopover>
+			</div>
+		);
+
+		const enhanceProtectionHeader = createInterpolateElement(
+			__(
+				'<b>Enhance protection</b> - Learn how to configure the firewall to inspect all requests and run before WordPress initializes',
+				'jetpack'
+			),
+			{
+				b: <strong />,
+			}
+		);
+
+		const bootstrapInstructions = (
+			<FoldableCard header={ enhanceProtectionHeader } className="waf__enhanced-protection">
+				<SettingsGroup hasChild>
+					<ul>
+						<li>
+							{ createInterpolateElement(
+								sprintf(
+									/* translators: Placeholder is the file path to the Firewall's bootstrap file. */
+									__(
+										'To ensure the firewall can best protect your site, please update: <code>auto_prepend_file</code> PHP directive to point to <code>%s</code> Typically this is set either in an .htaccess file or in the global PHP configuration; contact your host for further assistance.',
+										'jetpack'
+									),
+									this.props.bootstrapPath
+								),
+								{
+									code: <code />,
+								}
+							) }
+						</li>
+						<li>
+							{ __(
+								"Don't forget to undo this action when Firewall is turned off, or you uninstall Jetpack.",
```

Comment:
I just read that in the interface and was like "What? If I don't undo this when the firewall is turned off, I will uninstall Jetpack?" But what it actually means is that I should undo the action in either of those two cases, which does make sense :)

Therefore, small improvement suggestion.
```suggestion
								"Don't forget to undo this action when Firewall is turned off, or when you uninstall Jetpack.",
```

Reply:
That's a good point. Do you think we should have more proactive notices if someone deactivates the feature while leaving that in place or uninstall Jetpack?

admin notice if disabled with it still active?
e-mail to site admin if still active when Jetpack is deactivated/uninstalled?

Reply:
@kraftbj Actually, for the case when the firewall is turned off, we can be smart enough about it that the firewall does not actually do anything (this is not yet happening but should be easy to do, see p1650568832851519/1650548147.844039-slack-C029WFNV69M). For the other case I am not sure how exactly the deactivation/uninstall routine looks. If we do indeed remove all files, making this more prominent sounds like a good idea. Or at least create an empty file that does nothing in its place, if possible.